### ISSUE TITLE
Please revert latest DB backup changes

### ISSUE THREAD

**Issue by rbroeders** [domoticz/domoticz#4297]
Please revert the latest changes of the automatic DB backup. Since build 12248 the DB backup freezes domoticz for about 45 seconds. The result is that every hour
- web GUI is blocked for 45 seconds
- an error occurs that dzvents scripts take longer than 10 seconds
- multiple python plugins crash

Using build 12252 at this moment on a Pi 4 with ssd. Auto backup is to local ssd.

**Comment by gizmocuz** [comment]
@rbroeders ,

How big is your database ?

Could you try beta 12257, we now only 'sleep' when during a backup it encounters a busy or locked status

**Comment by rbroeders** [comment]
My DB is now 45Mb

I just updated to 12257, hope this helps.
Thanks

**Comment by gizmocuz** [comment]
@rbroeders , that is way way to large ! Something is not right there.
Best to open the database in a editor (for example SQLite Expert personal edition) and check which tables are so large, and find out why this happens
I have lot's of devices, and since 2012 my database is 20MB
I suspect 12257 will help, but that does not solve your issue, you might end up with an even larger database in some months

**Comment by rbroeders** [comment]
Thanks, I’ll check it tonight.

**Comment by rbroeders** [comment]
I just checked my monthly backups and the DB isn’t growing over the months.
I do have short log history set to 7 days and I have lots of THB sensors and a few light sensors logging lots of data.
I expect this is the cause of the size, but first want to check the result of 12257 without other changes.

**Comment by gizmocuz** [comment]
@rbroeders , that could indeed be one of the reasons, also check your lightinglog tables

**Comment by rbroeders** [comment]
With 12257 the backup took 3.3 seconds, so that’s great 👍 
After the backup I deleted the short log and now DB is 19 MB. I’ll look in the DB tonight to see if I can find some strange big tables.

Thanks for your help. The issue is solved :)

**Comment by rwaaren** [comment]
> After the backup I deleted the short log and now DB is 19 MB. I’ll look in the DB tonight to see if I can find some strange big tables.

There were at least two topics in the forum this year about very large databases that turned out to be caused by strange data in the Preferences table.  (Have not found a root cause 😒 ) 
Maybe it is a candidate to examine first ?

**Comment by rbroeders** [comment]
Did a check on the database and didn't find any huge tables. LightingLog was the biggest with around 100k rows, a few motion sensors and the 'pi-hole recent blocked' had about 15k row. I reduced the switches log retention from 30 to 14 days.
The biggest difference was the short log sensors, deleting them made a difference of 25MB. But this will grow back in the next week because I want to keep 7 days of log. 
Anyway, with your latest change the problem with the backups is solved so in my opinion a 45MB database shouldn't be a problem anymore.

Thanks again

**Comment by gizmocuz** [comment]
@rbroeders , 19MB is still a lot, if you compare that against my database of 19MB (including short log), and that is a database from 2016 till now
Could be you have lots of complicated floorplan drawings, large switch icons, extreme large data in user variables, maybe also check your 'scenelog' table

But do manually check your switchlog table, see if you have switch updates every few seconds, some bad designed plugins could cause this

**Comment by gizmocuz** [comment]
Glad it's solved, thanks for testing!

**Comment by rbroeders** [comment]
Checked the database size again and noticed that the backup of 00:00 was only 9MB. So I guess I should have done a shrink DB. Probably there's a cleaning routine at 00:00... After this the DB is growing with almost 200kB per hour.
Most of this is the short sensor log since I deleted that. And I deleted a few logs of switches and a text sensor which got big.
I did not discover any invalid logs due to bad plugins. Of course there's a lot of useless logs, but because there's only 1 general setting for log retention and not a per device override I can't optimize the logs.
